dig_test.go

```diff
@@ -1433,6 +1379,112 @@ func TestGroups(t *testing.T) {
 		require.Error(t, err, "failed to provide")
 		assert.Contains(t, err.Error(), "flatten can be applied to slices only")
 	})
+
+	t.Run("check providers for value groups aren't called if only they're provided", func(t *testing.T) {
+		type Param struct {
+			dig.In
+
+			Values []string `group:"foo,soft"`
+		}
+		type Result struct {
+			dig.Out
+
+			Value string `group:"foo"`
+		}
+		c := digtest.New(t)
+
+		c.RequireProvide(func() (Result, int) {
+			require.FailNow(t, "This shouldn't be called")
```

Comment:
nit: The majority of the tests in this file use "should not be called". Uniformity makes it easier to ctrl-f.